docs/source/cpp/streaming_execution.rst

```diff
@@ -305,3 +305,451 @@ Datasets may be scanned multiple times; just make multiple scan
+      {source}, cp::ConsumingSinkNodeOptions(consumer)));
+
+
+Consuming-Sink Example;
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: ConsumingSink Example)
+  :end-before: (Doc section: ConsumingSink Example)
+  :linenos:
+  :lineno-match:
+
+.. _stream_execution_order_by_sink_docs:
+
+``order_by_sink``
+-----------------
+
+``order_by_sink`` operation is an extension to the ``sink`` operation. 
+This operation provides the ability to guarantee the ordering of the 
+stream by providing the :class:`arrow::compute::OrderBySinkNodeOptions`. 
+Here the :class:`arrow::compute::SortOptions` are provided to define which columns 
+are used for sorting and whether to sort by ascending or descending values.
+
+Example::
+
+  arrow::compute::ExecNode *sink;
+
+  ARROW_ASSIGN_OR_RAISE(sink,
+  arrow::compute::MakeExecNode("order_by_sink", plan.get(),
+  {source}, 
+  arrow::compute::OrderBySinkNodeOptions{
+  /*sort_options*/arrow::compute::SortOptions{
+  {	arrow::compute::SortKey{
+  //Column key(s) to order by and how to order by these sort keys.
+  "a",
+  // Sort Order
+  arrow::compute::SortOrder::Descending 
+  }}},&sink_gen}));
+
+
+Order-By-Sink Example:
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: OrderBySink Example)
+  :end-before: (Doc section: OrderBySink Example)
+  :linenos:
+  :lineno-match:
+
+
+.. _stream_execution_select_k_docs:
+
+``select_k_sink``
+-----------------
+
+``select_k_sink`` option enables selecting k number of elements. 
+:class:`arrow::compute::SelectKOptions` which is a defined by 
+using :struct:`OrderBySinkNode` definition. This option returns a sink node that receives 
+inputs and then compute top_k/bottom_k.
+
+Create SelectK Option::
+
+  arrow::compute::SelectKOptions options = arrow::compute::SelectKOptions::TopKDefault(
+              /*k=*/2, {"i32"});
+
+  ARROW_ASSIGN_OR_RAISE(
+    arrow::compute::ExecNode * k_sink_node,
+    arrow::compute::MakeExecNode("select_k_sink",
+      plan.get(), {source},
+      arrow::compute::SelectKSinkNodeOptions{options, &sink_gen}));
+
+
+SelectK Example;
+
+.. literalinclude:: ../../../cpp/examples/arrow/execution_plan_documentation_examples.cc
+  :language: cpp
+  :start-after: (Doc section: KSelect Example)
+  :end-before: (Doc section: KSelect Example)
+  :linenos:
+  :lineno-match:
+
+.. _stream_execution_scan_docs:
+
+``scan``
+---------
+
+`scan` is an operation used to load and process data, and the behavior of is defined using 
+:class:`arrow::dataset::ScanNodeOptions`. This option contains a set of definitions. 
+The :ref:`dataset<cpp-dataset-reading>` API also use scanner for processing data.
+In contrast to `source` operation, `scan` operation can load the data and apply scanning
+operations like filter and project to the loaded data. 
```

Comment:
```suggestion
`scan` is an operation used to load and process datasets.  It should be preferred over the
more generic `source` node when your input is a dataset.  The behavior is defined using 
:class:`arrow::dataset::ScanNodeOptions`.  More information on datasets and the various
scan options can be found in :ref:`dataset<cpp-dataset-reading>`.

This node is capable of applying pushdown filters to the file readers which reduce
the amount of data that needs to be read.  This means you may supply the same
filter expression to the scan node that you also supply to the FilterNode because
the filtering is done in two different places.
```

Reply:
updated